src/ecdh.rs

```diff
@@ -132,11 +59,27 @@ impl SharedSecret {
         // The default `secp256k1_ecdh_hash_function_default` should always return 1.
         // and the scalar was verified to be valid(0 > scalar > group_order) via the type system
         debug_assert_eq!(res, 1);
-        ss.set_len(32); // The default hash function is SHA256, which is 32 bytes long.
-        ss
+        SharedSecret(buf)
+    }
+
+    /// Returns true if the underlying data buffer is empty.
+    pub fn is_empty(&self) -> bool {
+        self.0.is_empty()
```

Comment:
This will always return `false`, so is it really needed?

Reply:
Thanks, I incorrectly thought that all zeros would return true. Removed ... and schooled.